CONTRIBUTING.md

```diff
@@ -0,0 +1,257 @@
+
+### Code reviews
+
+All submissions, including submissions by project members, need to be reviewed before being merged.
+
+## Setup
+
+If you have not done so on this machine, you need to:
+ 
+* Install Git and configure your GitHub access
+* Install Java SDK (OpenJDK recommended)
+* Install [GraalVM](https://quarkus.io/guides/building-native-image)
+* Install platform C developer tools:
+    * Linux
+        * Make sure headers are available on your system (you'll hit 'Basic header file missing (<zlib.h>)' error if they aren't).
+            * On Fedora `sudo dnf install zlib-devel`
+            * Otherwise `sudo apt-get install libz-dev`
+    * macOS
+        * `xcode-select --install` 
+* Set `GRAALVM_HOME` to your GraalVM Home directory e.g. `/opt/graalvm` on Linux or `$location/JDK/GraalVM/Contents/Home` on macOS
+* Install Docker: it is used to run the integration tests for this project:
+    * Check [the installation guide](https://docs.docker.com/install/), and [the MacOS installation guide](https://docs.docker.com/docker-for-mac/install/)
+    * If you just install docker, be sure that your current user can run a container (no root required). 
+      On Linux, check [the post-installation guide](https://docs.docker.com/install/linux/linux-postinstall/)
+
+## Coding Guidelines
+
+### Java
+
+We follow the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html). See
+https://github.com/google/google-java-format for IDE plugins. The rules are not configurable.
+
+The build will fail if the code is not formatted. To format all files from the command line, run:
+ 
+```
+mvn fmt:format
+```
+
+Some aspects are not covered by the formatter: braces must be used with `if`, `else`, `for`, `do`
+and `while` statements, even when the body is empty or contains only a single statement.
+
+### XML
+
+The build will fail if XML files are not formatted correctly. Run the following command before you
+commit:
+
+```java
+mvn xml-format:xml-format
+```
+
+The formatter does not enforce a maximum line length, but please try to keep it below 100 characters
+to keep files readable across all mediums (IDE, terminal, Github...).
+
+## Coding style -- production code
+
+Do not use static imports. They make things harder to understand when you look at the code 
+someplace where you don't have IDE support, like Github's code view.
+
+Avoid abbreviations in class and variable names. A good rule of thumb is that you should only use
+them if you would also do so verbally, for example "id" and "config" are probably reasonable.
+Single-letter variables are permissible if the variable scope is only a few lines, or for commonly
+understood cases (like `i` for a loop index).
+
+Keep source files short. Short files are easy to understand and test. The average should probably 
+be around 200-300 lines. 
+
+### Javadoc
+
+All types in "API" packages must be documented. For "internal" packages, documentation is optional,
```

Comment:
Consider using another device instead of "<whatever>". Bold? Quote signs are not the best choice; quotation marks, like capital letters, tend to create a busy, distracting text and therefore should be avoided.

https://www.prismnet.com/~hcexres/textbook/high.html#:~:text=As%20a%20technical%20writer%2C%20limit,and%20therefore%20should%20be%20avoided.